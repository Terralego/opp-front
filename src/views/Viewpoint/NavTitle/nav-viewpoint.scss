@import "../../../variables";

.navTop {
  padding: 0;
  background-color: $darkblue;

  .navTop__go-back {
    background-color: $solidgray;

    button {
      background-color: $solidgray;
      background-image: none;
      height: 50px;

      > span {
        color: $white;
      }

      &:hover {
        text-decoration: none;
        background-color: #4a6479;
        color: #ffffff;
      }
    }
  }

  .navTop__viewpoint-name {
    color: $white;
    padding-left: 10px;
    background-color: $contrastedblue;

    h2 {
      margin-left: 10px;
      font-size: 28px;
      color: #FFFFFF;
      letter-spacing: 1px;
      text-align: left;
      font-weight: 300;
      text-transform: capitalize;
    }

    .icon {
      width: 22px;
      height: auto;
    }
  }

  .navTop__viewpoint-detail {
    a {
      .bp3-button {
        color: $white;

        .bp3-icon {
          color: $white;
        }
      }
    }
  }
}
