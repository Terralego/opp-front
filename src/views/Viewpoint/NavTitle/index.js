import React from 'react';
import { NavLink } from 'react-router-dom';

import { Navbar, NavbarGroup, Alignment, Button } from '@blueprintjs/core';

import './nav-viewpoint.scss';

export class NavTitle extends React.PureComponent {
  render() {
    const {
      viewpoint: { id, label },
      to,
      isNavBarVisible,
      toggleNavBar,
    } = this.props;
    return (
      <Navbar className="navTop">
        <NavbarGroup className="navTop__go-back" align={Alignment.LEFT}>
          <NavLink to={to.replace(/{{id}}/, id)}>
            <Button
              icon="arrow-left"
              onClick={() => !isNavBarVisible && toggleNavBar && toggleNavBar()}
              minimal
            />
          </NavLink>
        </NavbarGroup>
        <NavbarGroup className="navTop__viewpoint-name" align={Alignment.CENTER}>
          <img src="/images/icon_poi.svg" className="icon" alt="" />
          <Navbar.Heading>
            <h2>{label}</h2>
          </Navbar.Heading>
        </NavbarGroup>
      </Navbar>
    );
  }
}
export default NavTitle;
