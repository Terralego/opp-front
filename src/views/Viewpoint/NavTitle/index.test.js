import React from 'react';
import renderer from 'react-test-renderer';

import NavTitle from '.';

jest.mock('react-router-dom', () => ({
  NavLink: () => null,
}));

it('should render', () => {
  const data = {};
  const to = '';
  const tree = renderer.create((
    <NavTitle to={to} viewpoint={data} />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
