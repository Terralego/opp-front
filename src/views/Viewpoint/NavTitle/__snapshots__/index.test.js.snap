// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="bp3-navbar navTop"
>
  <div
    className="bp3-navbar-group bp3-align-left navTop__go-back"
  />
  <div
    className="bp3-navbar-group navTop__viewpoint-name"
  >
    <img
      alt=""
      className="icon"
      src="/images/icon_poi.svg"
    />
    <div
      className="bp3-navbar-heading"
    >
      <h2 />
    </div>
  </div>
</div>
`;
